sklearn/decomposition/_nmf.py

```diff
@@ -330,9 +330,14 @@ def _initialize_nmf(X, n_components, init=None, eps=1e-6,
                                                              copy=False)
         W = avg * rng.randn(n_samples, n_components).astype(X.dtype,
                                                             copy=False)
-        np.abs(H, out=H)
-        np.abs(W, out=W)
-        return W, H
+        # we do not write np.abs(H, out=H) to stay compatible with
+        # numpy 1.5 and earlier where the 'out' keyword is not
+        # supported as a kwarg on ufuncs
+        np.abs(H, H)
+        np.abs(W, W)
+        A = H.copy()
+        B = np.ones((n_components, n_features))
+        return W, H, A, B
```

Comment:
Docstring need to be updated (actually, I would rather not initialize A and B in this function, since it is the same for all init types, and even useless in the NMF case).